dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,601 @@
+
+        final Map<String, Param> initialParamsMap = ImmutableMap.of(
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true));
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        long time = System.currentTimeMillis();
+
+        final String serviceKey1 = String.format("all_lower_case_not_too_short_prefix_%d", time);
+        final String fileName = String.format("%s.yml", serviceKey1);
+        try(final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey1,
+                serviceKey1,
+                "Test-service-casing", false, initialParamsMap)) {
+
+            final Response serviceIntegrationResponse = serviceIntegrationResource
+                    .createServiceIntegration(request, response,
+                            createFormDataMultiPart(fileName, inputStream));
+            Assert.assertNotNull(serviceIntegrationResponse);
+            Assert.assertEquals(HttpStatus.SC_OK, serviceIntegrationResponse.getStatus());
+
+            final String serviceKeyCasingVariant1 = upperCaseRandom(serviceKey1, 30);
+
+            final Response serviceIntegrationByKey = serviceIntegrationResource
+                    .getServiceIntegrationByKey(request, response, serviceKeyCasingVariant1);
+            Assert.assertEquals(HttpStatus.SC_OK, serviceIntegrationByKey.getStatus());
+
+            final List<String> sites = new ArrayList<>();
+            final int max = 5;
+            for (int i = 1; i <= max; i++) {
+                sites.add(
+                        createServiceIntegrationParams(initialParamsMap, serviceKey1, request,
+                                response)
+                                .getIdentifier()
+                );
+            }
+
+            for (final String siteId : sites) {
+                final Response detailedIntegrationResponse = serviceIntegrationResource
+                        .getDetailedServiceIntegration(request, response,
+                                upperCaseRandom(serviceKey1, 30), siteId);
+                Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponse.getStatus());
+            }
+        }catch (Exception e){
+            Logger.error(ServiceIntegrationResourceTest.class, e);
+            fail();
+        }
+    }
+
+    @Test
+    public void Test_Required_Params() throws IOException, DotDataException, DotSecurityException {
+
+        final Map<String, Param> initialParamsMap = ImmutableMap.of(
+             "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true)
+        );
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        long time = System.currentTimeMillis();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'time' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)